### ISSUE TITLE
csv guess colum name detection behavior

### ISSUE THREAD

**Issue by y-ken** [embulk/embulk#162]
I have expected the column detection like below but It results c0, c1.... 
Would you please improve column name detection behavior?

```
# expected
    columns:
    - {name: ID, type: string}
    - {name: 名前, type: string}
    - {name: 住所, type: string}

# actual
    columns:
    - {name: c0, type: string}
    - {name: c1, type: string}
    - {name: c2, type: string}
```

## input files

All files encodings are utf-8.

```
$ cat << 'EOF' > test.csv
"ID","名前","住所"
"1","あ","北海道"
"2","い","東京"
EOF
```

```
$ cat << 'EOF' > partial-config.yml
in:
  type: file
  path_prefix: ./test.csv
out:
  type: stdout
EOF
```

## guess outputs

```
$ embulk guess partial-config.yml -o config.yml
2015-04-22 10:16:48.650 +0900: Embulk v0.6.2
2015-04-22 10:16:49.384 +0900 [INFO] (guess): Listing local files at directory '.' filtering filename by prefix 'test.csv'
2015-04-22 10:16:49.388 +0900 [INFO] (guess): Loading files [test.csv]
/Users/y-ken/bin/embulk!/embulk/guess/schema_guess.rb:81 warning: already initialized constant TRUE_STRINGS
/Users/y-ken/bin/embulk!/embulk/guess/schema_guess.rb:90 warning: already initialized constant FALSE_STRINGS
/Users/y-ken/bin/embulk!/embulk/guess/schema_guess.rb:98 warning: already initialized constant TYPE_COALESCE
exec: {}
in:
  type: file
  path_prefix: ./test.csv
  parser:
    charset: UTF-8
    newline: CRLF
    type: csv
    delimiter: ','
    quote: '"'
    escape: ''
    skip_header_lines: 0
    columns:
    - {name: c0, type: string}
    - {name: c1, type: string}
    - {name: c2, type: string}
out: {type: stdout}

Created 'config.yml' file.
```

**Comment by y-ken** [comment]
Adding an another case of related this issue.

```
$ vim test.csv
"No","種類","地域","市区町村コード"
1,"学生","北海道","01101"
2,"社会人","北海道","01102"
```

it results unexpected quoted column name.

```
exec: {}
in:
  type: file
  path_prefix: ./test.csv
  parser:
    charset: UTF-8
    newline: CRLF
    type: csv
    delimiter: ','
    quote: '"'
    escape: ''
    skip_header_lines: 1
    columns:
    - {name: '"No"', type: boolean}
    - {name: '"種類"', type: string}
    - {name: '"地域"', type: string}
    - {name: '"市区町村コード"', type: string}
out: {type: stdout}
```

**Comment by frsyuki** [comment]
Thank you for reporting! I fixed it at 75b6c70.

**Comment by y-ken** [comment]
input csv
```
"No","種類","地域","市区町村コード","都道府県名","市区町村名","地区名","最寄駅：名称","最寄駅：距離（分）","取引価格（総額）","坪単価","間取り","面積（㎡）","取引価格（㎡単価）","土地の形状","間口","延床面積（㎡）","建築年","建物の構造","建物の用途","前面道路：方位","前面道路：種類","前面道路：幅員（ｍ）","都市計画","建ぺい率（％）","容積率（％）","取引時点","備考"
1,"宅地(土地)","住宅地","40202","福岡県","大牟田市","大字岩本","吉野(福岡)","30分〜60分",2300000,"84000","","90",26000,"","","","","","","南東","市道",6.0,"第１種低層住居専用地域",40,60,"平成17年第３四半期",""
2,"宅地(土地)","宅地見込地","40202","福岡県","大牟田市","大字岩本","吉野(福岡)","30分〜60分",5200000,"43000","","390",13000,"ほぼ長方形","","","","","","","",,"市街化調整区域",60,200,"平成17年第３四半期",""
```

results
```
$ embulk guess partial-config.yml -o config.yml
2015-04-22 13:21:08.517 +0900: Embulk v0.6.4
2015-04-22 13:21:09.275 +0900 [INFO] (guess): Listing local files at directory '.' filtering filename by prefix '40_Fukuoka'
2015-04-22 13:21:09.279 +0900 [INFO] (guess): Loading files [40_Fukuoka Prefecture_20053.csv]
/Users/y-ken/bin/embulk!/embulk/guess/schema_guess.rb:81 warning: already initialized constant TRUE_STRINGS
/Users/y-ken/bin/embulk!/embulk/guess/schema_guess.rb:90 warning: already initialized constant FALSE_STRINGS
/Users/y-ken/bin/embulk!/embulk/guess/schema_guess.rb:98 warning: already initialized constant TYPE_COALESCE
undefined method `each' for nil:NilClass
    /Users/y-ken/bin/embulk!/embulk/guess/schema_guess.rb:39:in `types_from_array_records'
    file:/Users/y-ken/bin/embulk!/embulk/guess/csv.rb:63:in `guess_lines'
    file:/Users/y-ken/bin/embulk!/embulk/guess_plugin.rb:121:in `guess'
    file:/Users/y-ken/bin/embulk!/embulk/guess_plugin.rb:25:in `guess'
    Embulk$$GuessPlugin$$JavaAdapter_1181242988.gen:13:in `guess'
    file:/Users/y-ken/bin/embulk!/embulk/command/embulk_run.rb:319:in `run'
    classpath:embulk/command/embulk.rb:43:in `(root)'

Error: undefined method `each' for nil:NilClass
```
